Source/VMware.vSphereDSC/Classes/DatastoreBaseDSC.ps1

```diff
@@ -78,7 +79,35 @@ class DatastoreBaseDSC : VMHostEntityBaseDSC {
     Retrieves the Datastore with the specified name from the VMHost if it exists.
     #>
     [PSObject] GetDatastore() {
-        return Get-Datastore -Server $this.Connection -Name $this.Name -VMHost $this.VMHost -ErrorAction SilentlyContinue -Verbose:$false
+        $getDatastoreParams = @{
+            Server = $this.Connection
+            Name = $this.Name
+            VMHost = $this.VMHost
+            ErrorAction = 'SilentlyContinue'
+            Verbose = $false
+        }
+
+        $datastore = Get-Datastore @getDatastoreParams
+
+        <#
+        If the established connection is to a vCenter Server, Ensure is 'Present' and the Datastore does not exist on the specified VMHost,
+        we need to check if there is a Datastore with the same name on the vCenter Server.
+        #>
+        if ($this.Connection.ProductLine -eq $this.vCenterProductId -and $this.Ensure -eq [Ensure]::Present -and $null -eq $datastore) {
+            # We need to remove the filter by VMHost from the hashtable to search for the Datastore in the whole vCenter Server.
+            $getDatastoreParams.Remove('VMHost')
+
+            <#
+            If there is another Datastore with the same name on the vCenter Server but on a different VMHost, we need to inform the user that the Datastore cannot be created with the
+            specified name. Otherwise the Datastore will be created with appended '(1)' to its name.
```

Comment:
Change the comment. Explain VC accept s multiple datastore creation with sam name but changes the names internally to avoid name duplication. VC appends index '(<index number>)' to the datastore name.

Reply:
Fixed.